### ISSUE TITLE
Does the multi-label classifier optimize the entire model or it freezes BERT?

### ISSUE THREAD

**Issue by Hannabrahman** [kaushaltrivedi/fast-bert#188]
I wonder if the default code optimizes the entire model end-to-end or just the additional classifier layer parameters get updated?

Thanks

**Comment by aaronbriel** [comment]
I noticed the freeze/requires_grad logic is commented in learner_cls. @kaushaltrivedi how does one freeze the bert layers and only train the added custom layer?

**Comment by aaronbriel** [comment]
I tried adding a freeze_transformers_layer conditional in BertLearner's init function that set requires_grad to false for any param with the model_type in named_parameters but it didn't seem to have any effect. This approach worked for me in another implementation, resulting in exponentially reduced training times.

**Comment by aaronbriel** [comment]
https://github.com/kaushaltrivedi/fast-bert/pull/195

**Comment by lingdoc** [comment]
just for clarification, freezing the layers means that there is a single linear layer (i.e. classifier head?) being trained for classification rather than all the layers being used to update the weights (of the input text) during training?
